include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -374,7 +374,7 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
         // -> (U, v), (v, w), (w, x)
         // note that (x, t) is _not_ included but needs to be added later.
         BOOST_ASSERT(start_index <= end_index);
-        for (std::size_t i = start_index; i != end_index; ++i)
+        for (std::size_t i = start_index; i != end_index; (start_index < end_index ? ++i : --i))
```

Comment:
this fixes the case, I had the assertion in to try and generate a cucumber test and forgot to take it out again :(